examples/httpserver_simpletest_auto.py

```diff
@@ -0,0 +1,23 @@
+# SPDX-FileCopyrightText: 2022 Dan Halbert for Adafruit Industries
+#
+# SPDX-License-Identifier: Unlicense
+
+import socketpool
+import wifi
+
+from adafruit_httpserver import Server, Request, Response
+
+
+pool = socketpool.SocketPool(wifi.radio)
+server = Server(pool, "/static", debug=True)
+
+
+@server.route("/")
+def base(request: Request):
```

Comment:
This handler doesn't get called when you navigate to the device in a browser. For my device I'm loading `http://192.168.1.121/` but am getting a blank page without the message on it. In the dev tools on the browser I can see it's a 404 response. 

I was able to get it working by adding `[GET]` in the `@server.route` annotation arguments:

```
@server.route("/", [GET])
```

My assumption from these results is that methods argument for server route is functionally required. It doesn't raise an exception if you omit it, but will not ever get called since it won't match any methods. I did only basic testing on GET request so far though, perhaps the reality is more nuanced with other request types or scenarios. 

I think I am in favor of one of the following (but I don't have a strong opinion on which):

- Make `[GET]` the default value of the `methods` argument so that if the user doesn't declare it, they'll have default behavior that responds to GET requests. 

- Make the argument "fully required" and raise an exception if they do not pass this argument. This way the user won't have to troubleshoot the blank 404 responses if they end up in this situation. In this case, we'd also want to update any examples that have omitted this argument so that they'll run and serve responses as expected.

Reply:
It currently defaults to `GET`, but there is in fact a error in parsing logic, I identified the problem, will fix soon.

Thanks for spotting that.

Reply:
Ahh, I see. I tested it successfully in the latest commits. Thank you.